### DISCUSSION TITLE
otbr-agent failing with RadioSpinelNoResponse

### DISCUSSION THREAD

**Discussion by saramonteiro** [openthread/openthread#8764] [Q&A]
Hello everyone!

**- What am I trying to do and the issue I am facing:**

I am trying to build a OTBR using a Rasberry Pi 4 Model B + a [NXP k32W061 USB Dongle](https://www.nxp.com/products/wireless/multiprotocol-mcus/k32w-usb-dongle-for-bluetooth-le-zigbee-and-thread-networks:OM15080-K32W) and I am having the following error when running `journalctl -u otbr-agent.service -f`:

```
Feb 15 17:22:25 raspberrypi otbr-agent[933]: [CRIT]-PLAT----: HandleRcpTimeout() at ../../third_party/openthread/repo/src/lib/spinel/radio_spinel_impl.hpp:2169: RadioSpinelNoResponse
Feb 15 17:22:25 raspberrypi systemd[1]: otbr-agent.service: Main process exited, code=exited, status=6/NOTCONFIGURED
```

- **How I built everything:**

For building the OTBR I used the instructions provided here: https://openthread.io/guides/border-router/build 
For building the RCP for the Dongle I used the followed instructions from [here](https://github.com/openthread/ot-nxp/blob/main/src/k32w0/k32w061/README.md):

```
git clone --recurse-submodules https://github.com/openthread/ot-nxp
cd ot-nxp/
./script/bootstrap
sudo apt-get install python3-pip
pip3 install pycrypto
pip3 install pycryptodome
export NXP_K32W0_SDK_ROOT=<path to SDK>/SDK_2_6_9_K32W061DK6/
./script/build_k32w061
```

Then, I accessed _build_k32w061_ directory and I noticed that several bins were generated in different folders.
I was not sure which one should I use as RCP firmware then I tried _otc-cli-mtd.bin_ and _ot-rcp.bin_ for UART.
I am not sure but I suspect that the error from the otbr-agent is related to the miscommunication with the RPC`s Spinel.

I would appreciate some help,
thanks in advance! ⭐

**Comment by jwhui** [comment]
`ot-rcp.bin` is the one you want.

The `otbr-agent` error log you are seeing typically indicates that communication with the RCP is not working as expected.

I would reach out directly to NXP for support.

**Reply by saramonteiro** [reply]
ok, only to be more specific:

the ot-rcp.bin firmware which I used was from `ot-nxp/build_k32w061/rcp_only_uart_flow_control/openthread/examples/apps/ncp
` directory.

I am gonna check with NXP and come back to here until I have a fully working setup. Please, don`t close it until we come with a solution.

**Reply by jwhui** [reply]
If you have access to hardware from other vendors, you could try that as well.
